tiledb/tests/datatypes.py

```diff
@@ -43,14 +39,10 @@ def construct_from_string(cls, string):
 
 
 class RaggedArray(ExtensionArray):
-    def __init__(self, arrays, dtype=None):
-        if isinstance(dtype, RaggedDtype):
-            self._dtype = dtype
-            dtype = dtype.subtype
-        else:
-            self._dtype = RaggedDtype(dtype)
-
-        self._flat_arrays = [np.asarray(array, dtype=dtype) for array in arrays]
+    def __init__(self, arrays, dtype):
+        assert isinstance(dtype, RaggedDtype)
```

Comment:
Should we add a warning if `copy` is implemented in the superclass, so that we can remove it?

Reply:
Nevermind, it's already implemented in `RaggedArray` and no way to detect changes [here](https://github.com/holoviz/datashader/blob/615b659f79336e0c65e7f4d898a63aeed1645fcf/datashader/datatypes.py#L593).